docs/architecture/adr/ADR044-dev-namespaces.md

```diff
@@ -0,0 +1,74 @@
+# ADR044: Dev namespaces
+
+## Status
+
+Pending
+
+## Context
+
+Following the retirement of gsp-local the service teams have no way of testing
+changes before a merge to master. So the feedback cycle is very slow, and
+potentially risky. We need to enable devs to get more rapid feedback through use
+of namespaces that allow for deployments from arbitrary branches or from local
+`kubectl apply` actions.
+
+This presents several security problems that need to be addressed.
+
+The in-cluster concourse:
+
+* must not be able to create or edit daemonsets
```

Comment:
what's wrong with a daemonset?

Reply:
They describe pods that run on every node, which service teams shouldn't be interacting with.

Reply:
this discussion got [duplicated below](https://github.com/alphagov/gsp/pull/842#pullrequestreview-338109876), marking this one resolved